badge-maker/lib/badge-renderers.js

```diff
@@ -576,78 +576,110 @@ function forTheBadge({
+      LETTER_SPACING * message.length
+    : 0
+
+  // Compute horizontal layout.
+  // If a `labelColor` is set, the logo is always set against it, even when
+  // there is no label. When `needsLabelRect` is true, render a label rect and a
+  // message rect; when false, only a message rect.
+  const hasLabel = Boolean(label.length)
+  const needsLabelRect = hasLabel || (logo && labelColor)
+  let logoMinX, labelTextMinX
+  if (logo) {
+    logoMinX = LOGO_MARGIN
+    labelTextMinX = logoMinX + logoWidth + LOGO_TEXT_GUTTER
+  } else {
+    labelTextMinX = TEXT_MARGIN
+  }
+  let labelRectWidth, messageTextMinX, messageRectWidth
+  if (needsLabelRect) {
     if (hasLabel) {
-      labelWidth += 7
+      labelRectWidth = labelTextMinX + labelTextWidth + TEXT_MARGIN
     } else {
-      labelWidth -= 7
+      labelRectWidth = 2 * LOGO_MARGIN + logoWidth
     }
+    messageTextMinX = labelRectWidth + TEXT_MARGIN
+    messageRectWidth = 2 * TEXT_MARGIN + messageTextWidth
   } else {
-    labelWidth -= 11
+    if (logo) {
+      messageTextMinX = TEXT_MARGIN + logoWidth + LOGO_TEXT_GUTTER
+      messageRectWidth =
+        2 * TEXT_MARGIN + logoWidth + LOGO_TEXT_GUTTER + messageTextWidth
+    } else {
+      messageTextMinX = TEXT_MARGIN
+      messageRectWidth = 2 * TEXT_MARGIN + messageTextWidth
+    }
   }
 
-  // Add 20 px of padding, plus approximately 1.5 px of letter spacing per
-  // character.
-  messageWidth += 20 + 1.5 * message.length
-  const leftWidth = hasLogo && !hasLabel ? 0 : labelWidth
-  const rightWidth =
-    hasLogo && !hasLabel ? messageWidth + labelWidth : messageWidth
-
-  labelColor = hasLabel || hasLogo ? labelColor : color
-
-  color = escapeXml(color)
-  labelColor = escapeXml(labelColor)
-
-  let [leftLink, rightLink] = links
-  leftLink = escapeXml(leftLink)
-  rightLink = escapeXml(rightLink)
-  const { hasLeftLink, hasRightLink } = hasLinks({ links })
-
-  const accessibleText = createAccessibleText({ label, message })
+  // Render.
+  function renderLogo() {
+    return `
+      <image
+        x="${logoMinX}"
+        y="${0.5 * (BADGE_HEIGHT - LOGO_HEIGHT)}"
+        width="${logoWidth}"
+        height="${LOGO_HEIGHT}"
+        xlink:href="${escapeXml(logo)}"
+      />
+    `
+  }
 
   function renderLabelText() {
-    const { textColor } = colorsForBackground(labelColor)
-    const labelTextX = ((labelWidth + totalLogoWidth) / 2) * 10
-    const labelTextLength = (labelWidth - (24 + totalLogoWidth)) * 10
-    const escapedLabel = escapeXml(label)
-
-    const text = `<text fill="${textColor}" x="${labelTextX}" y="175" transform="scale(.1)" textLength="${labelTextLength}">${escapedLabel}</text>`
+    const { textColor } = colorsForBackground(outLabelColor)
+    const midX = labelTextMinX + 0.5 * labelTextWidth
+    const text = `
+      <text
+        transform="scale(.1)"
+        x="${10 * midX}"
+        y="175"
+        textLength="${10 * labelTextWidth}"
```

Comment:
would it be worth while adding constants for the scale or is it more readable inline/easier to hypothetically use different scale factor for the label and message/

Reply:
Sure, I think that's a nice idea.